test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -22,51 +22,199 @@
+                    frameVisibleLatch.countDown();
+                }
+            });
+
+            jTextField.addComponentListener(new ComponentAdapter() {
+                @Override
+                public void componentShown(ComponentEvent e) {
+                    super.componentShown(e);
+                    System.out.println("Component is visible + " + e.toString());
+                    componentVisibleLatch.countDown();
+                }
+            });
+
+            jTextField.addKeyListener(new KeyAdapter() {
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    super.keyPressed(keyEvent);
+                    int code = keyEvent.getKeyCode();
+                    int mod = keyEvent.getModifiersEx();
+                    if (code == '1' && mod == KeyEvent.SHIFT_DOWN_MASK) {
+                        keyPressedEventLatch.countDown();
+                        jLabel.setText("KEYPRESS received for Shift+1");
+                        System.out.println("KEYPRESS received for Shift+1");
+                    } else {
+                        jLabel.setText("Did not received KEY PRESS for Shift+1");
+                        System.out.println("Did not received KEY PRESS for Shift+1");
+                    }
+                }
+            });
+
+            Container container = frame.getContentPane();
+            container.add(jTextField, BorderLayout.SOUTH);
+            container.add(jLabel, BorderLayout.CENTER);
+            frame.setSize(300, 300);
+            frame.setLocationRelativeTo(null);
+            frame.setAlwaysOnTop(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.setVisible(true);
+        });
+
+        if (!isTopLevelVisible(frame, frameVisibleLatch)) {
+            throw new RuntimeException("Dialog is not visible after " + TIMEOUT + "  seconds");
+        }
+        System.out.println("frame is visible " + frame.toString());
+
+        if (!isJComponentVisible(jTextField, componentVisibleLatch)) {
+            throw new RuntimeException("Component is not visible after " + TIMEOUT + "   seconds");
+        }
+
+        Robot robot = new Robot();
+        robot.setAutoDelay(DELAY);
+        robot.waitForIdle();
+
+        SwingUtilities.invokeAndWait(() -> {
+            points[0] = jTextField.getLocationOnScreen();
+            rect[0] = jTextField.getBounds();
+        });
+
+        performMouseAction(robot, points[0].x + rect[0].width / 2, points[0].y + rect[0].height / 2);
+
+        // Press SHIFT + 1 keys
+        robot.waitForIdle();
+        robot.keyPress(KeyEvent.VK_SHIFT);
+        robot.keyPress(KeyEvent.VK_1);
+        robot.keyRelease(KeyEvent.VK_1);
+        robot.keyRelease(KeyEvent.VK_SHIFT);
+        robot.waitForIdle();
+
+        if (!keyPressedEventLatch.await(TIMEOUT, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Did not received KEY PRESS for Shift + 1 , test failed");
```

Comment:
```suggestion
            throw new RuntimeException("Did not receive KEY PRESS for Shift + 1 , test failed");
```
And the same for “KEY PRESS”.

Reply:
Done